### ISSUE TITLE
Wrong default momentum / clipping for SGD? 

### ISSUE THREAD

**Issue by eric-haibin-lin** [zackchase/mxnet-the-straight-dope#411]
In http://gluon.mxnet.io/chapter02_supervised-learning/linear-regression-gluon.html#Optimizer it says 

> Instead of writing stochastic gradient descent from scratch every time, we can instantiate a gluon.Trainer, passing it a dictionary of parameters. Note that the sgd optimizer in gluon actually uses SGD with momentum and clipping (both can be switched off if needed), since these modifications make it converge rather much better. We will discuss this later when we go over a range of optimization algorithms in detail.

I don't think this is correct. Does the trainer with SGD have default momentum? 
```
>>> net = gluon.nn.Dense(1)
>>> net.collect_params().initialize(mx.init.Normal(sigma=1.), ctx=mx.cpu())
>>> trainer = gluon.Trainer(net.collect_params(), 'sgd', {'learning_rate': 0.0001})
>>> trainer._optimizer
<mxnet.optimizer.SGD object at 0x7f9655005850>
>>> trainer._optimizer.momentum
0.0
```

**Comment by zackchase** [comment]
Hmm - thanks for the heads up. Not sure if my information is obsolete or was wrong all along.. Will fix now. 
So running now:
```
>>> trainer._optimizer.clip_gradient

>>>trainer._optimizer.momentum
0.0
```
Some random thoughts:
So default value for `clip_gradient` is `None`. Is this the most sensible default value? Vs what are the values it could take? The user can manually set it to 2 - will that actually run? Can the optimizer take separate clipping thresholds per weight matrix?  Does it make more sense to have two values - 1) boolean for whether clipping is turned on and 2) the magnitude at which to clip. How does the user specify which norm to clip by?

Note: the gluon API doc doesn't make this clear: http://mxnet.incubator.apache.org/api/python/gluon/gluon.html#mxnet.gluon.Trainer
